interface/main/main_screen.php

```diff
@@ -136,14 +156,17 @@ function generate_html_end()
     }
     $userid = $_SESSION['authId'];
     $form_response = empty($_POST['form_response']) ? '' : $_POST['form_response'];
-
     if ($form_response) {
         // TOTP METHOD enabled if TOTP is visible in post request
+        if (empty($_POST['totp']) && !$isU2F) {
+            $errormsg = xl("Please enter a valid code from the authenticator app");
+            $errortype = "TOTP";
+        }
```

Comment:
Did testing (this doesn't stop errors if $_POST['totp'] is empty when using totp. Recommend:
1. Remove the above code block.
2. Add require tag to following field: https://github.com/openemr/openemr/blob/63a64951be441f443a083362dc9833eac22f9a59/interface/main/main_screen.php#L284